File: lib/cfr/builtins/libs/flutter/painting/edgeInsetsDirectional.dart
import 'dart:core' as _fac9;
import 'dart:ui' as _a643;

import 'package:flutter/src/painting/basic_types.dart' as _487f;
import 'package:flutter/src/painting/edge_insets.dart' as _96d1;

import 'package:hydro_sdk/cfr/runtimeSupport.dart' as _36c2;

class VMManagedEdgeInsetsDirectional extends _36c2.VMManagedBox<_96d1.EdgeInsetsDirectional> {VMManagedEdgeInsetsDirectional({required this.table, required this.vmObject, required this.hydroState}) : super(
          table: table,
          vmObject: vmObject,
          hydroState: hydroState,
        ) { table  [
'start'
] = vmObject.start;
table  [
'top'
] = vmObject.top;
table  [
'end'
] = vmObject.end;
table  [
'bottom'
] = vmObject.bottom;
table  [
'getIsNonNegative'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { return [vmObject.isNonNegative,]; } );
table  [
'getFlipped'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { return [_36c2.maybeBoxObject<_96d1.EdgeInsetsDirectional>(object: vmObject.flipped, hydroState: hydroState, table: _36c2.HydroTable()),]; } );
table  [
'subtract'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { return [_36c2.maybeBoxObject<_96d1.EdgeInsetsGeometry>(object: vmObject.subtract(_36c2.maybeUnBoxAndBuildArgument<_96d1.EdgeInsetsGeometry, _fac9.dynamic>(luaCallerArguments  [
1
], parentState: hydroState)), hydroState: hydroState, table: _36c2.HydroTable()),]; } );
table  [
'add'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { return [_36c2.maybeBoxObject<_96d1.EdgeInsetsGeometry>(object: vmObject.add(_36c2.maybeUnBoxAndBuildArgument<_96d1.EdgeInsetsGeometry, _fac9.dynamic>(luaCallerArguments  [
1
], parentState: hydroState)), hydroState: hydroState, table: _36c2.HydroTable()),]; } );
table  [
'resolve'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { return [_36c2.maybeBoxObject<_96d1.EdgeInsets>(object: vmObject.resolve(_36c2.maybeUnBoxEnum(values: _a643.TextDirection.values, boxedEnum: luaCallerArguments  [
1
])), hydroState: hydroState, table: _36c2.HydroTable()),]; } );
table  [
'along'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { return [vmObject.along(_36c2.maybeUnBoxEnum(values: _487f.Axis.values, boxedEnum: luaCallerArguments  [
1
])),]; } );
table  [
'inflateSize'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { return [_36c2.maybeBoxObject<_a643.Size>(object: vmObject.inflateSize(_36c2.maybeUnBoxAndBuildArgument<_a643.Size, _fac9.dynamic>(luaCallerArguments  [
1
], parentState: hydroState)), hydroState: hydroState, table: _36c2.HydroTable()),]; } );
table  [
'deflateSize'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { return [_36c2.maybeBoxObject<_a643.Size>(object: vmObject.deflateSize(_36c2.maybeUnBoxAndBuildArgument<_a643.Size, _fac9.dynamic>(luaCallerArguments  [
1
], parentState: hydroState)), hydroState: hydroState, table: _36c2.HydroTable()),]; } );
table  [
'clamp'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { 
return [_36c2.maybeBoxObject<_96d1.EdgeInsetsGeometry>(object: vmObject.clamp(_36c2.maybeUnBoxAndBuildArgument<_96d1.EdgeInsetsGeometry, _fac9.dynamic>(luaCallerArguments  [
1
], parentState: hydroState), _36c2.maybeUnBoxAndBuildArgument<_96d1.EdgeInsetsGeometry, _fac9.dynamic>(luaCallerArguments  [
2
], parentState: hydroState)), hydroState: hydroState, table: _36c2.HydroTable()),]; } );
table  [
'toString'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { return [vmObject.toString(),]; } );
table  [
'getHorizontal'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { return [vmObject.horizontal,]; } );
table  [
'getVertical'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { return [vmObject.vertical,]; } );
table  [
'getCollapsedSize'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { return [_36c2.maybeBoxObject<_a643.Size>(object: vmObject.collapsedSize, hydroState: hydroState, table: _36c2.HydroTable()),]; } );
table  [
'getHashCode'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { return [vmObject.hashCode,]; } ); }

final _36c2.HydroTable table;

final _36c2.HydroState hydroState;

final _96d1.EdgeInsetsDirectional vmObject;

 }

void loadEdgeInsetsDirectional({required _36c2.HydroState hydroState, required _36c2.HydroTable table}) { 
table  [
'edgeInsetsDirectionalLerp'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { 

final returnValue = _96d1.EdgeInsetsDirectional.lerp(_36c2.maybeUnBoxAndBuildArgument<_96d1.EdgeInsetsDirectional?, _fac9.dynamic>(luaCallerArguments  [
1
], parentState: hydroState), _36c2.maybeUnBoxAndBuildArgument<_96d1.EdgeInsetsDirectional?, _fac9.dynamic>(luaCallerArguments  [
2
], parentState: hydroState), luaCallerArguments  [
3
]?.toDouble());if(returnValue!= null){return [_36c2.maybeBoxObject<_96d1.EdgeInsetsDirectional?>(object: returnValue, hydroState: hydroState, table: _36c2.HydroTable()),];}return []; } );
_36c2.registerBoxer<_96d1.EdgeInsetsDirectional>(boxer: ({required _96d1.EdgeInsetsDirectional vmObject, required _36c2.HydroState hydroState, required _36c2.HydroTable table}) { return  VMManagedEdgeInsetsDirectional(vmObject: vmObject, hydroState: hydroState, table: table); } ); } 

File: runtime/flutter/painting/edgeInsetsDirectional.ts
import { ISize } from "../../dart/ui/size";
import { TextDirection } from "../../dart/ui/textDirection";
import { Axis } from "./axis";
import { IEdgeInsets } from "./edgeInsets";
import { IEdgeInsetsGeometry } from "./edgeInsetsGeometry";
declare const flutter: {
    painting: {
        edgeInsetsDirectionalLerp: (
            a: IEdgeInsetsDirectional | undefined,
            b: IEdgeInsetsDirectional | undefined,
            t: number
        ) => IEdgeInsetsDirectional | undefined;
    };
};
export interface IEdgeInsetsDirectional {
    start: number;
    top: number;
    end: number;
    bottom: number;
    getIsNonNegative: () => boolean;
    getFlipped: () => IEdgeInsetsDirectional;
    subtract: (other: IEdgeInsetsGeometry) => IEdgeInsetsGeometry;
    add: (other: IEdgeInsetsGeometry) => IEdgeInsetsGeometry;
    resolve: (direction?: TextDirection | undefined) => IEdgeInsets;
    along: (axis: Axis) => number;
    inflateSize: (size: ISize) => ISize;
    deflateSize: (size: ISize) => ISize;
    clamp: (
        min: IEdgeInsetsGeometry,
        max: IEdgeInsetsGeometry
    ) => IEdgeInsetsGeometry;
    toString: () => string;
    getHorizontal: () => number;
    getVertical: () => number;
    getCollapsedSize: () => ISize;
    getHashCode: () => number;
}
export class EdgeInsetsDirectional implements IEdgeInsetsGeometry {
    public static zero = EdgeInsetsDirectional.only();
    public readonly start: number = undefined as any;
    public readonly top: number = undefined as any;
    public readonly end: number = undefined as any;
    public readonly bottom: number = undefined as any;
    public static lerp(
        a: IEdgeInsetsDirectional | undefined,
        b: IEdgeInsetsDirectional | undefined,
        t: number
    ): IEdgeInsetsDirectional | undefined {
        return flutter.painting.edgeInsetsDirectionalLerp(a, b, t);
    }
    private readonly _dart_getIsNonNegative: () => boolean = undefined as any;
    private readonly _dart_getFlipped: () => IEdgeInsetsDirectional =
        undefined as any;
    private readonly _dart_subtract: (
        other: IEdgeInsetsGeometry
    ) => IEdgeInsetsGeometry = undefined as any;
    private readonly _dart_add: (
        other: IEdgeInsetsGeometry
    ) => IEdgeInsetsGeometry = undefined as any;
    private readonly _dart_resolve: (
        direction?: TextDirection | undefined
    ) => IEdgeInsets = undefined as any;
    private readonly _dart_along: (axis: Axis) => number = undefined as any;
    private readonly _dart_inflateSize: (size: ISize) => ISize =
        undefined as any;
    private readonly _dart_deflateSize: (size: ISize) => ISize =
        undefined as any;
    private readonly _dart_clamp: (
        min: IEdgeInsetsGeometry,
        max: IEdgeInsetsGeometry
    ) => IEdgeInsetsGeometry = undefined as any;
    private readonly _dart_toString: () => string = undefined as any;
    private readonly _dart_getHorizontal: () => number = undefined as any;
    private readonly _dart_getVertical: () => number = undefined as any;
    private readonly _dart_getCollapsedSize: () => ISize = undefined as any;
    private readonly _dart_getHashCode: () => number = undefined as any;
    public getIsNonNegative(): boolean {
        return this._dart_getIsNonNegative();
    }
    public getFlipped(): IEdgeInsetsDirectional {
        return this._dart_getFlipped();
    }
    public subtract(other: IEdgeInsetsGeometry): IEdgeInsetsGeometry {
        return this._dart_subtract(other);
    }
    public add(other: IEdgeInsetsGeometry): IEdgeInsetsGeometry {
        return this._dart_add(other);
    }
    public resolve(direction?: TextDirection | undefined): IEdgeInsets {
        return this._dart_resolve(direction);
    }
    public along(axis: Axis): number {
        return this._dart_along(axis);
    }
    public inflateSize(size: ISize): ISize {
        return this._dart_inflateSize(size);
    }
    public deflateSize(size: ISize): ISize {
        return this._dart_deflateSize(size);
    }
    public clamp(
        min: IEdgeInsetsGeometry,
        max: IEdgeInsetsGeometry
    ): IEdgeInsetsGeometry {
        return this._dart_clamp(min, max);
    }
    public toString(): string {
        return this._dart_toString();
    }
    public getHorizontal(): number {
        return this._dart_getHorizontal();
    }
    public getVertical(): number {
        return this._dart_getVertical();
    }
    public getCollapsedSize(): ISize {
        return this._dart_getCollapsedSize();
    }
    public getHashCode(): number {
        return this._dart_getHashCode();
    }
}
